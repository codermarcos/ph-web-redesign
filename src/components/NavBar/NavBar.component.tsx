import Link from 'next/link';
import { useRouter } from 'next/router';

import {
  Menu,
  Nav,
} from '@/components/NavBar/NavBar.style';

function NavBar() {

  const router = useRouter();

  const active = router.asPath === '/'
    ? 'first'
    : 'last';

  return (
    <Nav active={active}>
      <Menu>
        <li>
          <Link href="/">
            Popular
          </Link>
        </li>

        <li>
          <Link href="/newest">
            Newest
          </Link>
        </li>
      </Menu>
    </Nav>
  );

}

export default NavBar;
