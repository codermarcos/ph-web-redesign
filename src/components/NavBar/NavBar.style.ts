import styled from 'styled-components';

import { px } from 'csx';

export const Nav = styled.nav({});

export const Menu = styled.menu({
  display: 'grid',
  gridTemplateColumns: '1fr 1fr',
});


