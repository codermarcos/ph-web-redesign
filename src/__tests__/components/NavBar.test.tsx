import { RenderResult, render } from '@testing-library/react';
import { useRouter } from 'next/router';

import NavBar from '@/components/NavBar/NavBar.component';

jest.mock(
  'next/router',
  () => ({
    useRouter: jest.fn(),
  })
);

describe(
  '[Component] -> components/NavBar',
  () => {

    // eslint-disable-next-line init-declarations
    let wrapper: RenderResult;

    beforeEach(
      () => {

        (useRouter as jest.Mock).mockReturnValue({ asPath: '/' });

        wrapper = render(
          <NavBar />
        );

      }
    );

    it(
      'should render correct header',
      () => {

        expect(wrapper.container).toMatchSnapshot();

      }
    );

  }
);
