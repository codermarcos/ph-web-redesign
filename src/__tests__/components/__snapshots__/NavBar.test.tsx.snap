// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] -> components/NavBar should render correct header 1`] = `
<div>
  <nav
    class="sc-ef123523-0 hfsmXA"
  >
    <menu
      class="sc-ef123523-1 ceKGjd"
    >
      <li>
        <a
          href="/"
        >
          Popular
        </a>
      </li>
      <li>
        <a
          href="/newest"
        >
          Newest
        </a>
      </li>
    </menu>
  </nav>
</div>
`;
